src/hooks/with-auth.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import { NextApiRequest, NextApiResponse, NextPage, NextPageContext } from 'next';
+import Router from 'next/router';
+
+import { NullableUserProfile, useUser } from './use-user';
+import { ISignInWithAuth0 } from '../instance';
+import { createLoginUrl } from '../utils/url-helpers';
+
+type RedirectToLoginProps = {
+  render: () => React.ReactElement;
+};
+
+class RedirectToLogin extends Component<RedirectToLoginProps> {
+  public constructor(props: RedirectToLoginProps) {
+    super(props);
+  }
+
+  public componentDidMount(): void {
+    window.location.assign(createLoginUrl(Router.pathname));
+  }
+
+  public render(): React.ReactElement {
+    return this.props.render();
+  }
+}
+
+type AuthenticatedProps = React.PropsWithChildren<{ user: NullableUserProfile }>;
+
+export default function withAuth(
+  InnerComponent: React.ElementType,
+  redirect: () => React.ReactElement,
```

Comment:
Can we go for an options object here fro the other params? (like https://github.com/auth0/auth0-react/blob/master/src/with-authentication-required.tsx#L75)

It gives us more flexibility when designing the API

Reply:
Yup, I'll change it.

Reply:
But if the instance will be providing the HOC, as in :

```js
// profile-ssr.jsx
import { withAuth } from '../lib/auth0';
```

does it still make sense to use an options object? Because it will only be two params now (the inner component, and the render prop).